nuxeo-mongodb-audit/src/main/java/org/nuxeo/mongodb/audit/pageprovider/MongoDBAuditPageProvider.java

```diff
@@ -0,0 +1,347 @@
+                }
+                // Convert Calendar objects
+                if (val[fidx] instanceof Calendar) {
+                    val[fidx] = ((Calendar) val[fidx]).getTime();
+                }
+            }
+
+            if (!isNonNullParam(val)) {
+                // skip predicate where all values are null
+                continue;
+            }
+
+            nbFilters++;
+
+            String op = predicate.getOperator();
+            if (op.equalsIgnoreCase("IN")) {
+
+                String[] values = null;
+                if (val[0] instanceof Iterable<?>) {
+                    List<String> l = new ArrayList<>();
+                    Iterable<?> vals = (Iterable<?>) val[0];
+
+                    for (Object v : vals) {
+                        if (v != null) {
+                            l.add(v.toString());
+                        }
+                    }
+                    values = l.toArray(new String[l.size()]);
+                } else if (val[0] instanceof Object[]) {
+                    values = (String[]) val[0];
+                }
+                filters.add(Filters.in(predicate.getParameter(), values));
+            } else if (op.equalsIgnoreCase("BETWEEN")) {
+                filters.add(Filters.gt(predicate.getParameter(), val[0]));
+                if (val.length > 1) {
+                    filters.add(Filters.lt(predicate.getParameter(), val[1]));
+                }
+            } else if (">".equals(op)) {
+                filters.add(Filters.gt(predicate.getParameter(), val[0]));
+            } else if (">=".equals(op)) {
+                filters.add(Filters.gte(predicate.getParameter(), val[0]));
+            } else if ("<".equals(op)) {
+                filters.add(Filters.lt(predicate.getParameter(), val[0]));
+            } else if ("<=".equals(op)) {
+                filters.add(Filters.lte(predicate.getParameter(), val[0]));
+            } else {
+                filters.add(Filters.eq(predicate.getParameter(), val[0]));
+            }
+        }
+
+        if (nbFilters == 0) {
+            return new Document();
+        }
+        return Filters.and(filters);
+    }
+
+    private Bson buildSearchFilter(String fixedPart, PredicateDefinition[] predicates,
+            DocumentModel searchDocumentModel) {
+        Document fixedFilter = Document.parse(fixedPart);
+        BsonDocument filter = buildFilter(predicates, searchDocumentModel).toBsonDocument(BsonDocument.class,
+                getMongoDBBackend().getAuditCollection().getCodecRegistry());
+        if (fixedFilter.isEmpty()) {
+            return filter;
+        } else if (filter.isEmpty()) {
+            return fixedFilter;
+        }
+        return Filters.and(fixedFilter, filter);
+    }
+
+    private boolean isNonNullParam(Object[] val) {
+        if (val == null) {
+            return false;
+        }
+        for (Object v : val) {
+            if (v != null) {
+                if (v instanceof String) {
```

Comment:
`instanceof` already does a `null` check so you can move down the `v != null` part in the final `else`